lib/morph.js

```diff
@@ -146,18 +146,20 @@ function updateTextarea (newNode, oldNode) {
 
 function updateSelect (newNode, oldNode) {
   if (!oldNode.hasAttributeNS(null, 'multiple')) {
+    var curChild = newNode.firstChild
     var i = 0
-    var curChild = oldNode.firstChild
+
     while (curChild) {
       var nodeName = curChild.nodeName
       if (nodeName && nodeName.toUpperCase() === 'OPTION') {
-        if (curChild.hasAttributeNS(null, 'selected')) break
+        if (curChild.hasAttributeNS(null, 'selected') && curChild.getAttributeNS(null, 'selected') !== 'null') break
         i++
       }
+
       curChild = curChild.nextSibling
     }
 
-    newNode.selectedIndex = i
+    newNode.selectedIndex = oldNode.selectedIndex = i
```

Comment:
In fact, if we don't update the newNode selectedIndex here, it keeps the value of the previous node (not sure why). 

For example in the test case:

```javascript
      var values = ['foo', 'bar']

      var a = html`<select>${values.map(function (e) { return html`<option value="${e}" selected="${e === 'bar' ? 'selected' : null}">${e}</option>` })}</select>`
      var b = html`<select>${values.map(function (e) { return html`<option value="${e}" selected="${e === 'foo' ? 'selected' : null}">${e}</option>` })}</select>`

      var res = nanomorph(a, b)
      t.equal(res.value, 'foo')
```

b.selectedIndex is `1`, but we choose `0`. If we don't update it, it stays that way and the resulting morph is wron. I do agree though that this should not be necessary. I think I've not enough knowledge about what goes down before we reach the `updateSelect` part to find out why...